src/tests/builder.rs

```diff
@@ -317,10 +317,12 @@ struct TmpEnvVar<'a> {
 }
 
 impl<'a> TmpEnvVar<'a> {
-    fn new(varname: &'static str) -> TmpEnvVar<'a> {
+    fn new(varname: &'static str, tmp_value: &'static str) -> TmpEnvVar<'a> {
```

Comment:
I agree with the change, I'd prefer `TmpEnvVar::new()` not to actually call `setenv()`. When possible, constructors should be side-effect-free.

The simplest fix would be to modify the body of `tmp_env_var()` to something like:

```rust
{
    let guard = TmpEnvVar::new(varname);
    env::set_var(varname, tmp_value);
    guard
}
```

Reply:
Given that `TmpEnvVar`'s sole purpose in life is to roll back a `set_env` call, maybe it might make sense to replace `new` with `TmpEnvVar::set_var` or something to that effect. I know this is test code, but the fact that `TmpEnvVar::new(...)` own *will* have a side effect when it is destroyed, it makes sense to relieve the possibility of surprise in this case by making it impossible to get into that situation.